pkgs/state-transition-runtime/build.zig

```diff
@@ -1,12 +1,30 @@
 const Builder = @import("std").Build;
 const std = @import("std");
 
+const zkvm_types = enum {
+    ceno,
+    powdr,
+    sp1,
+};
+
 pub fn build(b: *Builder) void {
     const target_query = .{ .cpu_arch = .riscv32, .os_tag = .freestanding, .abi = .none, .cpu_model = .{ .explicit = &std.Target.riscv.cpu.generic_rv32 } };
 
     const target = b.resolveTargetQuery(target_query);
     const optimize = b.standardOptimizeOption(.{});
 
+    // Declare the -Dzkvm option, which is a choice between all supported zkvms
+    const zkvm = b.option(zkvm_types, "zkvm", "zkvm target") orelse .powdr;
+    const zkvm_module = b.addModule("zkvm", .{
```

Comment:
ok so we are inserting the target zkvm interface here that we have wrapped in each of individual folders?

Reply:
not quite: depending on the `-Dzkvm=` option we passed to `zig build`, it will use the files corresponding to the correct zkvm. So it's only using one subdirectory at a time.

I want to expand the whole thing so that all the binaries are built and can be released, but this is not necessary at this stage.

Reply:
yes, this doesn't need to be part of this build but proving manager's build can trigger these builds one by one to generate the target binaries

> depending on the -Dzkvm= option we passed to zig build, it will use the files corresponding to the correct zkvm. So it's only using one subdirectory at a time

right what I meant was that one particular zk vm gets wrapped into "zkvm" module which is used by the zig `main`